miden-tx/tests/integration/bench/tx_benchmark.rs

```diff
@@ -1,16 +1,133 @@
 use crate::MockDataStore;
-use miden_tx::TransactionExecutor;
+use miden_lib::transaction::ToTransactionKernelInputs;
+use miden_objects::{accounts::AccountStub, transaction::TransactionArgs};
+use miden_tx::{TransactionExecutor, TransactionHost};
+use vm_processor::{
+    AdviceExtractor, AdviceInjector, AdviceProvider, ExecutionError, ExecutionOptions, Host,
+    HostResponse, ProcessState, RecAdviceProvider,
+};
+
+// CONSTANTS
+// ================================================================================================
+
+const END_OF_PROLOGUE: u8 = 0;
+const END_OF_NODE_PROCESSING: u8 = 1;
+const END_OF_TX_SCRIPT_PROCESSING: u8 = 2;
+const END_OF_EPILOGUE: u8 = 3;
```

Comment:
I would use the same scheme here as we do for events and error codes - i.e., the first 16 bits are set to 2 and the remaining 16 bits is a counter that increments (0, 1, 2 etc.).